grouper/api/handlers.py

```diff
@@ -147,6 +155,34 @@ def get(self, name=None):
             })
 
 
+class BulkUsers(GraphHandler):
+    """API endpoint for bulk retrieval of user data.
+
+    This returns the same information as the Users and ServiceAccounts endpoints, but supports
+    multiple returning the data of multiple users to save on API call overhead.
+    """
+    def get(self):
+        names = self.get_argument("names", None)
+        if names is None:
+            usernames = self.graph.user_metadata.iterkeys()
+        else:
+            usernames = names.split(",")
+
+        cutoff = int(self.get_argument("cutoff", 100))
+        include_service_accounts = self.get_argument("include_role_users", "no") == "yes"
+
+        with self.graph.lock:
+            data = {}
+            for name in usernames:
+                try:
+                    data[name] = get_individual_user_info(
+                        self, name, cutoff,
+                        service_account=None if include_service_accounts else False)
+                except NoSuchUser:
+                    return self.notfound("User/Service Account ({}) not found.".format(name))
```

Comment:
seems potentially undesirable from a client POV for a single invalid username to tank the request. the patterns i've seen the most is to not include the invalid username in the result. open to other suggestions.